raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -867,6 +878,10 @@ private boolean handleVoteResponse(
         VoteResponseData.PartitionData partitionResponse =
             response.topics().get(0).partitions().get(0);
 
+        if (partitionResponse.preVote()) {
+            throw new UnsupportedOperationException("PreVote=true responses are not supported yet");
+        }
```

Comment:
I would remove this. If the replica is not sending preVote set to true then it should just assume that the response will not have the preVote field set to true.